pkg/route/routeapihelpers/routeapihelpers.go

```diff
@@ -42,3 +44,30 @@ func IngressURI(route *routev1.Route, host string) (*url.URL, *routev1.RouteIngr
 	}
 	return nil, nil, fmt.Errorf("no ingress for host %s in route %s in namespace %s", host, route.ObjectMeta.Name, route.ObjectMeta.Namespace)
 }
+
+// ValidateHost checks that a route's host name satisfies DNS requirements, with
+// the assumption that the caller has already checked for an empty host name.
+// Unless the allowNonCompliant annotation is set to true, host name must have
+// at least two labels, with each label no more than 63 characters from the set of
+// alphanumeric characters, '-' or '.', and must start and end with an alphanumeric
+// character. A trailing dot is allowed. The total host name length must be no more
+// than 253 characters.
+// If allowNonCompliant is set to true, it uses a smaller set of conditions from
+// IsDNS1123Subdomain, e.g. character set as described above, and total host name
+// length must be no more than 253 characters.
+func ValidateHost(host string, allowNonCompliant string, specPath *field.Path) field.ErrorList {
+	result := field.ErrorList{}
+
+	if allowNonCompliant == "true" {
+		errs := kvalidation.IsDNS1123Subdomain(host)
+		if len(errs) != 0 {
+			result = append(result, field.Invalid(specPath.Child("host"), host, fmt.Sprintf("host must conform to DNS naming conventions: %v", errs)))
```

Comment:
It would make sense for the caller to pass in the full path to the field that is being validated (i.e., `spec.host`).  Then `ValidateHost` would use that `*field.Path` value directly instead of using `specPath.Child("host")` here and below.